confmap/confmap_test.go

```diff
@@ -150,17 +150,17 @@ func TestUnmarshalWithIgnoreUnused(t *testing.T) {
 		"string":  "this is a string",
 	}
 	conf := NewFromStringMap(stringMap)
-	require.Error(t, conf.Unmarshal(&TestIDConfig{}))
-	assert.NoError(t, conf.Unmarshal(&TestIDConfig{}, WithIgnoreUnused()))
+	require.Error(t, conf.Unmarshal(&testIDConfig{}))
+	assert.NoError(t, conf.Unmarshal(&testIDConfig{}, WithIgnoreUnused()))
 }
 
-type TestConfig struct {
-	Boolean   *bool              `mapstructure:"boolean"`
-	Struct    *Struct            `mapstructure:"struct"`
-	MapStruct map[string]*Struct `mapstructure:"map_struct"`
+type testConfig struct {
+	Boolean   *bool                `mapstructure:"boolean"`
+	Struct    *myStruct            `mapstructure:"struct"`
```

Comment:
Not sure about this. I think to correctly do the right thing for unmarshalling the struct needs to be public.

Reply:
The fields need to be public, however the struct themselves can be unexported.